### PR TITLE
Add left/top and right/bottom track segments so they can be styled independently

### PR DESCRIPTION
**Pull request by mgwalker**
This adds left and right track segments that can be targeted by CSS (with .slider-track-left and .slider-track-right, respectively).  It allows users to style the "unselected" part of the slider, which is useful if you're building a "red-amber-green" style slider.

jsfiddle: http://jsfiddle.net/qfjjv2vp/

### CODE REVIEW

**Comment by seiyria** [comment]
This looks pretty nice, I gotta say. Would you make some tests for situations where you color each of these components, as well as verify the defaults are correct?

**Comment by mgwalker** [comment]
I'll be glad to make some tests, but I'm honestly not sure how to go about writing tests for this.  It seems like it'll be a manual thing since you've actually got to look at the rendered slider to verify it's correct.  Do you just want a sample added?

I promise I'm not trying to be a pain, I'm just not sure how to give you what you're asking for.  :)

**Comment by seiyria** [comment]
Sure, no worries. Basically, create the slider as you have in your tests, and just verify that, after creation, the respective slider tracks actually have their colors set appropriately (select them, check their color).

**Comment by mgwalker** [comment]
So, here's what I'm guessing:  I add input elements to SpecRunner.tpl, then add spec that creates the sliders, selects them, and checks the colors?  Since I'll need CSS to style them, is there somewhere I can put a CSS file for testing (I assume you don't want it inlined in the SpecRunner.tpl)?

**Comment by seiyria** [comment]
You can probably just inline it; I believe that's how the other slider CSS is handled. But yeah, that's the gist of it. That's also how the example page is done.

**Comment by mgwalker** [comment]
I added some tests that verify the background color as well as the left and right segment widths.  Hopefully this is what you had in mind.  :)

**Comment by seiyria** [comment]
Hey, it looks great to me. @rovolution - what do you think?

**Comment by mgwalker** [comment]
Incidentally, I just found #199.  This could satisfy @KANekT's needs as well, so that's cool.

**Comment by rovolution** [comment]
@mgwalker @seiyria just finished reviewing. This is a really well-done PR...thanks!

One more thing before I merge this in...could you add an example to the ```/tpl/index.tpl``` file?

It can be something simple where it shows how you can use CSS to style different parts of the track

**Comment by mgwalker** [comment]
Sure thing, I'll do it first thing Monday morning.

**Comment by mgwalker** [comment]
Added a sample to index.tpl.  Let me know if there's anything else I should do!

**Comment by rovolution** [comment]
@mgwalker one final thing...could you add those styles to our .sass file as well?

https://github.com/seiyria/bootstrap-slider/blob/master/sass/slider.scss

other than that, this looks great!

**Comment by mgwalker** [comment]
Sure.  I'm curious why you have both SASS and LESS, though.  Seems like extra effort to maintain, but no gain.  That's a discussion for another place, though.

**Comment by seiyria** [comment]
Same reason tools like Bootstrap do it; it lets the users use whichever tool they're most comfortable with go down that route.

**Comment by mgwalker** [comment]
Fair enough.  I've updated the SASS to match the LESS.

**Comment by rovolution** [comment]
@mgwalker merging in! thanks for the contribution!

File: sass/slider.scss [hunk 1]
```diff
@@ -110,6 +115,13 @@ $baseBorderRadius: 4px;
 	@include border-radius($baseBorderRadius);
 }
 
+.slider-track-left, .slider-track-right {
+  position: absolute;
+  background: transparent;
+  .box-sizing(border-box);
```
Comment (jamesgary) [inline sass/slider.scss]: We're using grunt and sass, and this line gives us the following error when including this file:

```
Line 121 of node_modules/bootstrap-slider/sass/slider.scss: Invalid CSS after "  .box-sizing": expected "{", was "(border-box);"
```

Is this valid sass?

Comment (mgwalker) [inline sass/slider.scss]: Apparently not.  It should probably be

```@include box-sizing(border-box);```

I've never used SASS before and it's not in the default Grunt task, so I probably never ran it.

**Comment by rovolution** [comment]
@jamesgary just hotfixed

**Comment by jamesgary** [comment]
Thanks! :two_hearts:

**Comment by rovolution** [comment]
@mgwalker As I was thinking about this, I thought make more sense to call these class tags ```slider-track-low``` and ```slider-track-high``` versus left and right. The reason being, lets say you have a situation where the user has a vertically-aligned slider. The value range makes more sense versus the direction, which always depends on the orientation.

If its not too much trouble, would you be able to make those changes?

**Comment by mgwalker** [comment]
Yes, I've create a new pull request in #305 for the changes.

**Pull request merged**